R/utils.R

```diff
@@ -178,7 +178,7 @@ Linter <- function(fun, name = linter_auto_name()) { # nolint: object_name.
     stop("`fun` must be a function taking exactly one argument.", call. = FALSE)
   }
   force(name)
-  structure(fun, class = "linter", name = name)
+  structure(fun, class = c("linter", "function"), name = name)
```

Comment:
Why add "function" as an explicit class?

Reply:
my understanding is that's the reason for the CRAN warning -- in all.equal(linter(), ...) S3 dispatch uses all.equal.default, with this change it will correctly use all.equal.function

Reply:
Oh I see.